cdp-agentkit-core/src/actions/cdp/webhooks.ts

```diff
@@ -0,0 +1,127 @@
+
+const CREATE_WEBHOOK_PROMPT = `
+Create a new webhook to receive real-time updates for on-chain events. 
+Supports monitoring wallet activity or smart contract events by specifying:
+- Callback URL for receiving events
+- Event type (wallet_activity, smart_contract_event_activity, erc20_transfer or erc721_transfer)
+- Addresses to monitor
+Also supports monitoring erc20_transfer or erc721_transfer, when those are defined at least one of these filters needs to be provided (only one of them is required):
+- Contract address to listen for token transfers
+- Sender address for erc20_transfer and erc721_transfer events (listen on transfers originating from this address))
+- Recipient address for erc20_transfer and erc721_transfer events (listen on transfers being made to this address))
+`;
+
+// Define the webhook event types
+const WebhookEventType = z.enum([
+  "wallet_activity",
+  "smart_contract_event_activity",
+  "erc20_transfer",
+  "erc721_transfer",
+]);
+
+const WebhookNetworks = z.enum(["base-mainnet", "base-sepolia"]);
+
+// Create a flexible event filters schema
+const EventFilters = z.array(
+  z
+    .object({
+      from_address: z.string().optional().describe("Sender address for token transfers"),
+      to_address: z.string().optional().describe("Recipient address for token transfers"),
+      contract_address: z.string().optional().describe("Contract address for token transfers"),
+    })
+    .refine(
+      data => {
+        // Ensure at least one filter is provided
+        return Object.keys(data).length > 0;
+      },
+      { message: "At least one filter must be provided" },
+    ),
+);
+
+const EventTypeFilter = z.object({
+  addresses: z.array(z.string()).describe("List of wallet or contract addresses to monitor"),
+});
+
+/**
+ * Input schema for create webhook action.
+ */
+export const CreateWebhookInput = z.object({
+  notificationUri: z.string().url().describe("The callback URL where webhook events will be sent"),
+  eventType: WebhookEventType,
+  eventTypeFilter: EventTypeFilter,
+  eventFilters: EventFilters.optional(),
+  networkId: WebhookNetworks,
+});
+
+/**
+ * Creates a new webhook for monitoring on-chain events
+ *
+ * @param wallet - The wallet used to create the webhook
+ * @param args - Object with arguments needed
+ * @returns Details of the created webhook
+ */
+export async function createWebhook(
+  wallet: Wallet,
+  args: z.infer<typeof CreateWebhookInput>,
+): Promise<string> {
+  try {
+    const { notificationUri, eventType, eventTypeFilter, eventFilters, networkId } = args;
+
+    const webhookOptions: CreateWebhookOptions = {
+      networkId,
+      notificationUri,
+      eventType,
+    };
+
+    // Handle different event types with appropriate filtering
+    switch (eventType) {
+      case "wallet_activity":
+        webhookOptions.eventTypeFilter = {
+          addresses: eventTypeFilter.addresses || [],
+          wallet_id: "", // this is required by SDK, but can be an empty value
```

Comment:
Why is this required by the SDK if it is unused? 

Can we delete line 84 entirely or must it be specified as `""`

Reply:
SDK requires this param although it's not required, something we'll look into, but we need this param for now.